### ISSUE TITLE
Add Support for `typescript-eslint` v8

### ISSUE THREAD

**Issue by Kenneth-Sills** [iamturns/eslint-config-airbnb-typescript#354]
Tomorrow, the ESLint Typescript Plugin's v8 will be released via [PR 9165](https://github.com/typescript-eslint/typescript-eslint/pull/9165). The changes for this are described in the [prerelease blog post](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta) (though that may come to be outdated on release), and we'll need corresponding updates for our users to stay up-to-date.

This WILL be a breaking change due to updates in required rules, the API exposed by the new version (which may break siblings plugins), and the move of all formatting rules to the separate [ESLint Stylistic](https://eslint.style/) project.

- [ ] Migrate `@typescript-eslint/no-throw-literal` to `@typescript-eslint/only-throw-error`.
- [ ] Remove `@typescript-eslint/no-loss-of-precision` (deprecated TS rule, handled correctly in base ESLint).
- [ ] Remove or replace formatting Rules.
- Leaving `no-unused-vars` unchanged *should* be fine. This would keep us in line with the AirBnB base config and the justification text of the standard makes no indication that caught errors are excluded.

This does NOT address #331 (supporting ESLint v9), which must still wait for [upstream compatibility](https://github.com/airbnb/javascript/issues/2961). Nor will it add first-class support for flat configuration.

**Comment by Kenneth-Sills** [comment]
Wonderful, the release is out and includes [an updated blog post](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8)! I don't immediately see additional changes needed from the list above.

**Comment by emilgoldsmith** [comment]
I get this because of the removal of stylistic errors:

> TypeError: Key "rules": Key "@typescript-eslint/lines-between-class-members": Could not find "lines-between-class-members" in plugin "@typescript-eslint". Did you mean "@/lines-between-class-members"

**Comment by Kenneth-Sills** [comment]
Yup! The full list of old rules moved to the stylistic plugin can be found [here](https://github.com/typescript-eslint/typescript-eslint/blob/2761ca83becc0b8c6d76a14fb2d2465a1d9c45a5/packages/eslint-plugin/tests/docs.test.ts#L128).

I hope to have a PR for the maintainer this weekend, but in the interim an example MVP project setup would look something like:

<details>
<summary>Details</summary>

*Warning: I slapped this together to use as a reference, just copy and pasting this probably won't work.*

```json
// package.json
{
    "dependencies": {
		"@eslint/eslintrc": "^3.1.0",
        "eslint-config-airbnb-typescript": "^18.0.0",
		"typescript-eslint": "^8.0.0"
    },
    "overrides": {
        "eslint-config-airbnb-typescript": {
            "@typescript-eslint/parser": "^8.0.0",
            "@typescript-eslint/eslint-plugin": "^8.0.0"
        }
    }
}
```

```js
// eslint.config.js

import { FlatConfig } from '@eslint/eslintrc';
import typescriptPlugin from 'typescript-eslint';

const compat = new FlatCompat({
    baseDirectory: import.meta.dirname,
});

export default typescriptPlugin.config(
    ...compat.extends('airbnb-base'),
    ...compat.extends('airbnb-typescript/base'),
	...typescriptPlugin.configs.recommendedTypeChecked,
	{
		rules: {
		    "@typescript-eslint/lines-between-class-members": "off",

			// Already disabled by `prettier-eslint`, if you're using that.
			"@typescript-eslint/brace-style": "off",
		    "@typescript-eslint/comma-dangle": "off",
		    "@typescript-eslint/comma-spacing": "off",
		    "@typescript-eslint/func-call-spacing": "off",
		    "@typescript-eslint/keyword-spacing": "off",
		    "@typescript-eslint/no-extra-parens": "off",
		    "@typescript-eslint/no-extra-semi": "off",
		    "@typescript-eslint/object-curly-spacing": "off",
		    "@typescript-eslint/semi": "off",
		    "@typescript-eslint/space-before-blocks": "off",
		    "@typescript-eslint/space-before-function-paren": "off",
		    "@typescript-eslint/space-infix-ops": "off",
		    "@typescript-eslint/quotes": 0,
		},
	},
);
```

</details>

**Comment by DiFuks** [comment]
@Kenneth-Sills It seems to me that the fix should not only consist of disabling removed rules. Airbnb-typescript should support migrating these rules to similar ones from eslint stylistic. In our organization's config, we have done so. But maybe I am mistaken and the maintainers have a different view on this

**Comment by Kenneth-Sills** [comment]
@DiFuks Absolutely! The config above was really to get people going in the interim, and I had yet to have a look at the `@stylistic` ruleset to know the migration. A PR has been submitted via #356 now, which migrates all rules appropriately and just explains how to disable them in the updated README.

Feel free to take a look and let me know if you see any issues, I always appreciate a second set of eyes!

**Comment by SteveW94** [comment]
As this currently blocks updating to typescript-eslint 8 which was released: Any news here? :)

**Comment by Kenneth-Sills** [comment]
@SteveW94 The maintainer doesn't have much activity, they may be on hiatus. A bit busy this week, but if I do have some time this weekend I can take a look at forking / starting a new project.

**Comment by khaleelkhalifa** [comment]
> @SteveW94 The maintainer doesn't have much activity, they may be on hiatus. A bit busy this week, but if I do have some time this weekend I can take a look at forking / starting a new project.

I would be interested in this too
